tests/components/todoist/test_calendar.py

```diff
@@ -107,3 +108,64 @@ async def test_calendar_custom_project_unique_id(todoist_api, hass, state):
 
     state = hass.states.get("calendar.all_projects")
     assert state.state == "off"
+
+
+async def test_async_get_events_no_due_date(api, task):
+    """Test an event without a due date is ignored."""
+    task.due = None
+    api.get_tasks.return_value = [task]
+    project_data = {"name": "test", "id": "12345"}
+    todoist_project_data = TodoistProjectData(project_data, labels=[], api=api)
+
+    start_date = datetime(2022, 1, 1, 0, 0, 0, tzinfo=timezone.utc)
+    end_date = datetime(2022, 1, 2, 0, 0, 0, tzinfo=timezone.utc)
+    events = await todoist_project_data.async_get_events(start_date, end_date)
```

Comment:
I realize that some of the tests may already do this style of test, however our design standards have the tests not due typical unit testing, but instead go through the APIs. For example, an acceptable way to exercise this is through the rest API (google calendar and calendar platform have some examples of testing this way). The idea is to do a little bit more of an end-to-end test to have more confidence when making system wide changes that may affect many components, and make them a bit easier to maintain when internal components are refactored.

Reply:
Thanks, I'll take a look at some of the calendar tests for an example of integration testing this functionality.

Reply:
It looked like the Google calendar tests were mocking the API responses, which wouldn't test the intended code path.  I unfortunately don't have a lot of free time at the moment to figure out a better way to write these tests so I've removed them.  Perhaps proper tests could be added for this component in a future PR.

Reply:
Well, the part i was thinking about was just calling the rest API. If you already had it working to return data from that function, then the same set up should work when being called from the Rest APIs. Google just happens to be mocking google calendar API responses, but that isn't the part you necessarily need to follow.

Reply:
Since this was a force push i can't see the old changes, otherwise would be happy to help contribute to make it work.

Reply:
By the way, it seems like the changes keep getting forced pushed so I have to re-review the entire PR since the history is gone and I can't just review what changed since my last review.

Reply:
Sorry about that, I was following the [docs](https://developers.home-assistant.io/docs/development_catching_up/) but I guess I don't need to do that with every push.

Reply:
> Since this was a force push i can't see the old changes, otherwise would be happy to help contribute to make it work.

These were the tests that were removed: https://github.com/home-assistant/core/pull/79481/commits/a2bf9a9c0295a0479066335ea89ca844874cb329#diff-b97322e720d3864f9b6b0ed3f95d28858bd3b654b3c5d7c58b68779084862386